pennylane_orquestra/orquestra_device.py

```diff
@@ -0,0 +1,384 @@
+"""
+# pylint: disable=protected-access, consider-using-enumerate
+
+import abc
+import json
+import re
+import uuid
+
+from pennylane import QubitDevice
+from pennylane.operation import Expectation, Tensor
+from pennylane.ops import Identity
+from pennylane.utils import decompose_hamiltonian
+from pennylane.wires import Wires
+
+from pennylane_orquestra._version import __version__
+from pennylane_orquestra.cli_actions import loop_until_finished, qe_submit, write_workflow_file
+from pennylane_orquestra.gen_workflow import gen_expval_workflow
+from pennylane_orquestra.utils import _terms_to_qubit_operator_string
+
+
+class OrquestraDevice(QubitDevice, abc.ABC):
+    """The Orquestra base device.
+
+    Provides the :meth:`~.execute` and :meth:`~.batch_execute` methods which allows
+    remote device executions by generating, submitting Orquestra workflows and
+    processing their results.
+
+    The :meth:`~.batch_execute` method can be utilized to send workflows that
+    contain several circuits which are computed in parallel on a remote device.
+
+    The workflow files generated are placed into a user specific data folder
+    specified by the output of ``appdirs.user_data_dir("pennylane-orquestra",
+    "Xanadu")``. By default, such files are removed (see
+    ``keep_files`` keyword argument). After each device execution,
+    filenames for the generated workflows are stored in the ``filenames``
+    attribute.
+
+    Computing the expectation value of the identity operator does not involve a
+    workflow submission (hence no files are created).
+
+    Args:
+        wires (int, Iterable[Number, str]]): Number of subsystems represented
+            by the device, or iterable that contains unique labels for the
+            subsystems as numbers (i.e., ``[-1, 0, 2]``) or strings (``['ancilla',
+            'q1', 'q2']``). Default 1 if not specified.
+        shots (int): number of circuit evaluations/random samples used to estimate
+            expectation values of observables
+        analytic (bool): If ``True``, the device calculates expectation values
+            analytically. If ``False``, a finite number of samples set by the
+            argument ``shots`` are used to estimate these quantities.
+
+    Keyword Args:
+        backend=None (str): the Orquestra backend device to use for the
+            specific Orquestra backend, if applicable
+        batch_size=10 (int): the size of each circuit batch when using the
+            :meth:`~.batch_execute` method to send multiple workflows
+        keep_files=False (bool): Whether or not the workflow files
+            generated during the circuit execution should be kept or deleted.
+        resources=None (dict): An option for Orquestra, specifies the resources
+            provisioned for the clusters running each workflow step.
+        timeout=300 (int): The maximum time until a job will timeout after getting no
+            response from Orquestra (in seconds).
```

Comment:
(minor) There seems to be a mix of capitalized-with-punctuation and not here. Maybe keep to one or the other; perhaps non-capitalized if only a single sentence? :thinking:

Reply:
Fair point, updated! Thanks